### ISSUE TITLE
VCS install overwrites unrelated local editable install

### ISSUE THREAD

**Issue by partfile** [pypa/pipenv#1752]
After installing `'-e .' --dev` a subsequent install from VCS (another random package) changes the virtualenv easy-install.pth so that it no longer points at `.` 

<details><summary>$ python -m pipenv.help output</summary>
Pipenv version: `'11.8.0'`

Pipenv location: `'/usr/local/lib/python3.6/site-packages/pipenv'`

Python location: `'/usr/local/opt/python/bin/python3.6'`

Other Python installations in `PATH`:

  - `2.7`: `/usr/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.6`: `/usr/local/bin/python3.6m`
  - `3.6`: `/usr/local/bin/python3.6`
  - `3.6`: `/usr/local/bin/python3.6`

  - `2.7.10`: `/usr/bin/python`
  - `3.6.4`: `/usr/local/bin/python3`
  - `3.6.4`: `/usr/local/bin/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.6.4',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.4.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST '
                     '2017; root:xnu-4570.41.2~1/RELEASE_X86_64',
 'python_full_version': '3.6.4',
 'python_version': '3.6',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `TERM_PROGRAM`
  - `SHELL`
  - `TERM`
  - `TMPDIR`
  - `TERM_PROGRAM_VERSION`
  - `TERM_SESSION_ID`
  - `USER`
  - `SSH_AUTH_SOCK`
  - `PATH`
  - `PWD`
  - `LANG`
  - `XPC_FLAGS`
  - `XPC_SERVICE_NAME`
  - `SHLVL`
  - `HOME`
  - `LOGNAME`
  - `DISPLAY`
  - `SECURITYSESSIONID`
  - `OLDPWD`
  - `EDITOR`
  - `ZSH`
  - `PAGER`
  - `LESS`
  - `LC_CTYPE`
  - `LSCOLORS`
  - `LS_COLORS`
  - `_`
  - `__CF_USER_TEXT_ENCODING`
  - `PIP_PYTHON_PATH`
  - `PYTHONUNBUFFERED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/usr/local/git/man:/Users/partfile/Library/Python/2.7/bin:/Users/partfile/.phabricator/arcanist/bin:/usr/local/opt/fzf/bin`
  - `SHELL`: `/bin/zsh`
  - `EDITOR`: `/usr/local/bin/nvim`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/Users/partfile/Desktop/piperr`

---------------------------

Contents of `Pipfile` ('/Users/partfile/Desktop/piperr/Pipfile'):

```toml
[[source]]

url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"


[packages]

snips_nlu = {git = "ssh://git@github.com/snipsco/snips-nlu.git"}


[dev-packages]

"e1839a8" = {path = ".", editable = true}


[requires]

python_version = "3.6"

```

Contents of `Pipfile.lock` ('/Users/partfile/Desktop/piperr/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "25a679a500b6ec02ec0be929acf45421bf5eef93ed0292b298f6766f388bc934"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "3.6"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.python.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "piperr": {
            "editable": true,
            "git": "git@github.com:partfile/piperr.git",
            "ref": "575f65dcb91d3fc476cadbfadb3cd84136a4041d"
        },
        "snips-nlu": {
            "git": "ssh://git@github.com/snipsco/snips-nlu.git"
        }
    },
    "develop": {
        "e1839a8": {
            "editable": true,
            "path": "."
        }
    }
}

```
</details>

------------

##### Expected result

Expected to be able to add a VCS package, and continue editing my local project. 

##### Actual result

The effect here is that my local changes are no longer picked up - it is using a copy of my code in the venv src/ directory

##### Steps to replicate

Note: my working directory is a git repo, and I had already pushed a commit to remote. I'm not certain that is a factor, but I suspect it may be.

    $ pipenv install '-e .' —dev
    $ pipenv --venv
    /Users/partfile/.local/share/virtualenvs/piperr-51TwRZDX
    $ cat /Users/partfile/.local/share/virtualenvs/piperr-51TwRZDX/lib/python3.6/site-packages/easy-install.pth 
    /Users/partfile/Desktop/piperr

Good so far! but....

    $ pipenv install git+ssh://git@github.com/snipsco/snips-nlu.git#egg=snips_nlu
    $ cat /Users/partfile/.local/share/virtualenvs/piperr-51TwRZDX/lib/python3.6/site-packages/easy-install.pth
     /Users/partfile/.local/share/virtualenvs/piperr-51TwRZDX/src/piperr

And that directory is a copy of my local project, but at the last committed revision ..

**Comment by partfile** [comment]
Any thoughts on this? Please let me know if I can describe the issue better.

**Comment by techalchemy** [comment]
@partfile that file simply tracks things you've installed in your environment. looks like you've reset your environment at some point to me. Just run `pipenv lock` and then `pipenv sync` and you should be fine

**Comment by partfile** [comment]
well, ok - just fyi `pipenv sync --dev` does seem to fix the pth for me. It will then claim that the VCS install is there (pip freeze shows it) but it is broken.

If I make the VCS install editable, then lock & sync again, things are ok, the pth file is fine (contains the src version of the VCS + the local path to the project I'm working on)

NOTE that sync is not really discussed in the documentation, so I'm not sure how I would have know that it would help.

I tried several times, uninstalling things, then re-installing, and the VCS install still does that weird thing to the local-editable (pointing to a VCS version of local-editable in the pth file). Only after the sync are things right again. Still smells like a bug to me, or at the very least a strangely complicated process (broken until you sync?).

**Comment by techalchemy** [comment]
`Sync` is just the second half of `update`. Since you keep talking about the contents of your `.pth` file it’s quite hard to follow what’s actually wrong here. Is there some behavior you encountered that gave you an error? I need to see error messages or logs or problems of some kind beyond the contents of a file representing setuptools internals

**Comment by partfile** [comment]
Ah ok - so my issue description was unclear. Apologies for that.

The actual error I encountered is that I would make changes in my working directory, but they weren't picked up when I was running `pipenv run <my_code>`

The `.pth` file contents alter what is on my python path when I'm working in a pipenv environment. After installing `'-e .' --dev` the `.pth` file had `/Users/partfile/Desktop/piperr` ... this is correct, it means that my working directory is on my python path.

After installing a random VCS package, the `.pth` file had `/Users/partfile/.local/share/virtualenvs/piperr-51TwRZDX/src/piperr` ... so my working directory was no longer in my python path.

It looks to me like pipenv looked at the `git remote` from my working directory, copied that into that `src` directory, and the put that on my path instead of my actual working directory.

**Comment by techalchemy** [comment]
The reason the issue is unclear is because pipenv doesn’t copy things to your src dir or touch the pth file directly

**Comment by partfile** [comment]
ok, well maybe it's a virtualenv issue then. Thanks for the `sync` suggestion.
